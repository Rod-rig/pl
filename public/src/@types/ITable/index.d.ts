export type OrderType = 'asc' | 'desc';

export type RangeType = number[] | null;

export interface ITableProps {
  order: OrderType;
  sort: string;
  chars: string[];
  range?: RangeType;
}

export interface ITable extends ITableProps {
  id?: string;
  isLoaded: boolean;
  table: object[];
  url: string;
  sortHandler(e: Event): void;
}
