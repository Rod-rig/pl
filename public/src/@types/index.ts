export * from './ITable';
export * from './IMatch';
export * from './ITournament';
