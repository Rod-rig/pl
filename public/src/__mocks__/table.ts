export default {
  table: [
    {
      'd': 3,
      'goals against': 21,
      'goals for': 88,
      'l': 1,
      'points': 84,
      'position': 1,
      'teamName': 'Manchester City',
      'w': 27,
    },
    {
      'd': 5,
      'goals against': 23,
      'goals for': 60,
      'l': 5,
      'points': 68,
      'position': 2,
      'teamName': 'Manchester United',
      'w': 21,
    },
    {
      'd': 9,
      'goals against': 35,
      'goals for': 75,
      'l': 4,
      'points': 66,
      'position': 3,
      'teamName': 'Liverpool',
      'w': 19,
    },
  ],
};
