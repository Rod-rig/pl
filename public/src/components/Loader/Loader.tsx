import { CircularProgress, createStyles, Theme } from "@material-ui/core";
import { withStyles } from "@material-ui/core/styles";
import * as React from "react";

const styles = ({ spacing }: Theme) =>
  createStyles({
    root: {
      left: "50%",
      marginLeft: spacing(-5 / 2),
      marginTop: spacing(-5 / 2),
      position: "absolute",
      top: "50%",
    },
  });

export const Loader = withStyles(styles)((props: any) => (
  <CircularProgress classes={{ root: props.classes.root }} />
));
