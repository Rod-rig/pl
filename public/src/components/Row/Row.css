@import '../../css/_variables.css';

.row .row__cell:not(:nth-child(2)) {
    text-align: center;
}

@media (--md-breakpoint) {
    .row .row__cell {
        padding-left: 6px;
        padding-right: 6px;
    }

    .row__cell:nth-child(3),
    .row__cell:nth-child(7),
    .row__cell:nth-child(8) {
        display: none;
    }
}

@media (--sm-breakpoint) {
    .row__cell:nth-child(4),
    .row__cell:nth-child(5),
    .row__cell:nth-child(6) {
        display: none;
    }

    .row .row__cell:last-child {
        padding-right: 6px;
    }
}
