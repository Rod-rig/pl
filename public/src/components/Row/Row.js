import React from 'react';

const Row = (props) => {
    return (
        <tr>
            <td>{props.row.position}</td>
            <td>{props.row.teamName}</td>
            <td>{props.row.matches}</td>
            <td>{props.row.w}</td>
            <td>{props.row.d}</td>
            <td>{props.row.l}</td>
            <td>{props.row.goals}</td>
            <td>{props.row.points}</td>
        </tr>
    );
};

export default Row;
