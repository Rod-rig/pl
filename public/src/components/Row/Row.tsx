import {Avatar, TableCell, TableRow, withStyles} from '@material-ui/core';
import * as React from 'react';

export interface IRow {
  logo?: string;
  [index: string]: any;
}

interface IProps {
  chars: string[];
  classes?: any;
  logo?: string;
  row: IRow;
}

const decorate = withStyles(({breakpoints, palette, spacing}) => ({
  cell: {
    '&:last-child': {
      [breakpoints.down('sm')]: {
        paddingRight: 0.75 * spacing.unit,
      },
    },
    'text-align': 'center',
    [breakpoints.down('sm')]: {
      paddingLeft: 0.75 * spacing.unit,
      paddingRight: 0.75 * spacing.unit,
    },
  },
  draw: {
    [breakpoints.down('xs')]: {
      display: 'none',
    },
  },
  edge: {
    '& td': {
      borderColor: palette.grey[50],
    },
    'backgroundColor': palette.grey[300],
  },
  goals_against: {
    [breakpoints.down('sm')]: {
      display: 'none',
    },
  },
  goals_for: {
    [breakpoints.down('sm')]: {
      display: 'none',
    },
  },
  img: {
    '& img': {
      'object-fit': 'contain',
    },
    'border-radius': 0,
    'display': 'inline-block',
    'height': 25,
    'marginRight': 5,
    'verticalAlign': 'middle',
    'width': 25,
  },
  info: {
    alignItems: 'center',
    display: 'flex',
  },
  loss: {
    [breakpoints.down('xs')]: {
      display: 'none',
    },
  },
  middle: {
    backgroundColor: palette.grey[200],
  },
  played: {
    [breakpoints.down('sm')]: {
      display: 'none',
    },
  },
  row: {
    backgroundColor: palette.common.white,
  },
  // shortName: {
  //   display: 'inline',
  //   [breakpoints.up('sm')]: {
  //     display: 'none',
  //   },
  // },
  // team: {
  //   [breakpoints.down('xs')]: {
  //     display: 'none',
  //   },
  // },
  teamName: {
    'text-align': 'left',
    [breakpoints.down('xs')]: {
      width: '100%',
    },
  },
  top: {
    backgroundColor: palette.grey[400],
  },
  win: {
    [breakpoints.down('xs')]: {
      display: 'none',
    },
  },
}));

const renderAvatar = (logo: string, altText: string, className: string) => {
  return logo ? (
    <Avatar src={logo} alt={altText + '\'s logo'} className={className}/>
  ) : undefined;
};

const highlightCell = (stage: string) => {
  if (stage === 'Champions League' || stage === 'Champions League Qualification' || stage === 'Relegation') {
    return 'edge';
  } else if (stage === 'Europa League' || stage === 'Europa League Qualification' || stage === 'Playoffs') {
    return 'middle';
  } else {
    return 'row';
  }
};

export const Row = decorate((props: IProps) => {
  const {classes, row} = props;
  const rowClass = highlightCell(row.current_outcome);
  const topClass = row.rank === 1 && row.current_outcome !== 'Playoffs' ? classes.top : '';
  const rowProps = {
    className: `${classes[rowClass]} ${topClass}`,
    hover: !row.current_outcome,
  };
  return (
    <TableRow {...rowProps}>
      {
        props.chars.map((val, i): JSX.Element => {
          if (val !== 'team') {
            return (
              <TableCell
                key={i}
                padding='checkbox'
                className={`${classes.cell} ${classes[val] ? classes[val] : ''}`}
              >
                {row[val]}
              </TableCell>
            );
          } else {
            return (
              <TableCell key={i} padding='checkbox' className={`${classes.cell} ${classes.teamName}`}>
                <div className={classes.info}>
                  {renderAvatar(row.logo, row[val], classes.img)}
                  <span className={classes.team}>{row.team.name}</span>
                  {/*<span className={classes.shortName}>{row.shortName}</span>*/}
                </div>
              </TableCell>
            );
          }
        })
      }
    </TableRow>
  );
});
