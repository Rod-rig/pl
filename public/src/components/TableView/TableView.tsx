import {
  createStyles,
  Paper,
  Table,
  TableBody,
  Theme,
  Typography,
  withStyles,
  WithStyles,
} from "@material-ui/core";
import { observer } from "mobx-react";
import * as React from "react";
import { Loader, Nav, Row, TableHeadView } from "../";
import { IGroup, ITable, ITeam } from "../../@types";

const styles = ({ breakpoints, palette, spacing, typography }: Theme) =>
  createStyles({
    group: {
      ...typography.h6,
      backgroundColor: palette.secondary.main,
      color: palette.common.white,
      fontSize: "",
      padding: spacing(2, 3),
    },
    header: {
      backgroundColor: palette.primary.main,
      color: palette.primary.contrastText,
      padding: spacing(1),
      [breakpoints.up("lg")]: {
        margin: spacing(0, 3),
        padding: spacing(1, 3, 3),
      },
    },
    paper: {
      marginBottom: spacing(1),
      [breakpoints.up("lg")]: {
        margin: spacing(0, 3, 3),
      },
    },
    title: {
      margin: "0.35em 0",
      [breakpoints.up("sm")]: {
        fontSize: "3.5em",
      },
    },
  });

interface IProps extends WithStyles<typeof styles> {
  store: ITable;
}

export const TableView = withStyles(styles)(
  observer(
    class extends React.Component<IProps, {}> {
      public render() {
        const { classes, store } = this.props;

        return store.isLoaded ? (
          <React.Fragment>
            <div className={classes.header}>
              {store.title && (
                <Typography
                  variant="h4"
                  className={classes.title}
                  color="inherit"
                >
                  {store.title}
                </Typography>
              )}
              <Nav />
            </div>
            {store.table.map((group: IGroup, key: number) => {
              return (
                <Paper key={key} className={classes.paper}>
                  {group.name && (
                    <div className={classes.group}>Group {group.name}</div>
                  )}
                  <Table>
                    <TableHeadView
                      order={store.table[key].order}
                      sortName={store.table[key].sortName}
                      sortHandle={store.sortHandler.bind(store, key)}
                      chars={store.chars}
                    />
                    <TableBody>
                      {group.team_standings.map(
                        (row: ITeam, index: number): JSX.Element => (
                          <Row key={index} row={row} chars={store.chars} />
                        ),
                      )}
                    </TableBody>
                  </Table>
                </Paper>
              );
            })}
          </React.Fragment>
        ) : (
          <Loader />
        );
      }
    },
  ),
);
