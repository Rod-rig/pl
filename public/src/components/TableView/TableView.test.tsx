import {shallow} from 'enzyme';
import * as React from 'react';
import {ITable} from '../../@types/ITable';
import table from '../../__mocks__/table';
import {Loader} from '../Loader/Loader';
import {Row} from '../Row';
import TableView from './TableView';

describe('TableView', () => {
  const mockStore: ITable = {
    chars: ['position', 'teamName', 'points'],
    isLoaded: true,
    order: 'asc',
    sort: '',
    sortHandler: () => {
      return;
    },
    table: table.table,
    url: '',
  };
  const tableElement = shallow(<TableView store={mockStore}/>);
  const notRenderedTable = shallow(<TableView store={{...mockStore, ...{isLoaded: false}}}/>);

  it('should render rows correctly', () => {
    const row = tableElement.find(Row);
    expect(row).toHaveLength(table.table.length);
    expect(row.first().prop('row')).toEqual(table.table[0]);
  });

  it('should render loader', () => {
    expect(notRenderedTable.find(Loader)).toHaveLength(1);
  });
});
