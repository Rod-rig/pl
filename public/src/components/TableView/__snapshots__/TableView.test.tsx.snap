// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableView should render correctly with default props 1`] = `"<div class=\\"class_1-header-2\\"><h4 class=\\"MuiTypography-root class_1-title-4 MuiTypography-h4 MuiTypography-colorInherit\\">Title</h4><a class=\\"MuiButtonBase-root MuiButton-root withRouter()-containedSecondary-35 MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/tournament/undefined\\"><span class=\\"MuiButton-label\\">Table</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiButton-root withRouter()-containedSecondary-35 MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/fixtures/undefined\\"><span class=\\"MuiButton-label\\">Fixtures</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiButton-root withRouter()-containedSecondary-35 MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/results/undefined\\"><span class=\\"MuiButton-label\\">Results</span><span class=\\"MuiTouchRipple-root\\"></span></a><a class=\\"MuiButtonBase-root MuiButton-root withRouter()-containedSecondary-35 MuiButton-contained MuiButton-containedSecondary MuiButton-sizeSmall\\" tabindex=\\"0\\" role=\\"button\\" aria-disabled=\\"false\\" href=\\"/predictions?tournament_id=undefined\\"><span class=\\"MuiButton-label\\">Prediction</span><span class=\\"MuiTouchRipple-root\\"></span></a></div><div class=\\"MuiPaper-root MuiPaper-elevation1 class_1-paper-3 MuiPaper-rounded\\"><div class=\\"class_1-group-1\\">Group A</div><table class=\\"MuiTable-root\\"><thead class=\\"MuiTableHead-root\\"><tr class=\\"MuiTableRow-root MuiTableRow-head\\"><th class=\\"MuiTableCell-root Component-cell-86  MuiTableCell-head MuiTableCell-paddingCheckbox\\" aria-sort=\\"ascending\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root Component-cell-86  MuiTableCell-head MuiTableCell-paddingCheckbox\\" aria-sort=\\"ascending\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" title=\\"Sort\\"><svg class=\\"MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th><th class=\\"MuiTableCell-root Component-cell-86  MuiTableCell-head MuiTableCell-paddingCheckbox\\" aria-sort=\\"ascending\\" scope=\\"col\\"><span class=\\"MuiButtonBase-root MuiTableSortLabel-root\\" tabindex=\\"0\\" role=\\"button\\" title=\\"Sort\\">points<svg class=\\"MuiSvgIcon-root MuiTableSortLabel-icon MuiTableSortLabel-iconDirectionDesc\\" focusable=\\"false\\" viewBox=\\"0 0 24 24\\" aria-hidden=\\"true\\" role=\\"presentation\\"><path d=\\"M20 12l-1.41-1.41L13 16.17V4h-2v12.17l-5.58-5.59L4 12l8 8 8-8z\\"></path></svg></span></th></tr></thead><tbody class=\\"MuiTableBody-root\\"><tr class=\\"MuiTableRow-root Component-edge-137 Component-top-147\\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">95</td></tr><tr class=\\"MuiTableRow-root Component-edge-137 \\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">30</td></tr><tr class=\\"MuiTableRow-root Component-edge-137 \\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">0</td></tr><tr class=\\"MuiTableRow-root Component-middle-143 \\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">30</td></tr><tr class=\\"MuiTableRow-root Component-middle-143 \\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">31</td></tr><tr class=\\"MuiTableRow-root Component-row-145  MuiTableRow-hover\\"><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135 Component-teamName-146 MuiTableCell-body MuiTableCell-paddingCheckbox\\"></td><td class=\\"MuiTableCell-root Component-cell-135  MuiTableCell-body MuiTableCell-paddingCheckbox\\">25</td></tr></tbody></table></div>"`;
